test/plugin/base_test.rb

```diff
@@ -954,11 +962,54 @@ def test_http_request_skip_setting_latency
       end
       verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
         assert_equal HTTP_REQUEST_MESSAGE, entry['httpRequest'], entry
-        assert_nil get_fields(entry['jsonPayload'])['httpRequest'], entry
+        fields = get_fields(entry['jsonPayload'])
```

Comment:
Let's not change the behavior here. If `latency` comes in as invalid format, we want to throw away the data.

If we change the order of the code in the `set_log_entry_fields` to delete the field first before trying to cast it, this test should pass without being modified. Something like:

```
            value = fields.delete(original_key) # Delete it first. If the casting later returns nil, we just throw away the invalid data.
            next if value.nil?
            casted_value = send(cast_fn, value)
            next if casted_value.nil?
            extracted_fields[destination_key] = casted_value
```